stdlib/public/core/StringComparable.swift

```diff
@@ -164,10 +164,12 @@ extension _StringGuts {
     //
     // FIXME: String ordering should be consistent across all platforms.
     if left.isASCII && right.isASCII {
-      defer { _fixLifetime(left); _fixLifetime(right) }
```

Comment:
All that being said, I think this pattern should be improving soon according to @atrick, @aschwaighofer :-)

Reply:
@milseman I'm not sure where "all that being said" was said. What was your motivation for rewriting the `defer` idiom? At -O there should be no difference at all. If you see one, we need a bug. At -Onone, there are unnecessary retain/release pairs in both cases. They have to do with the fact that fixLifetime is generic. The (noescape) closure captures are guaranteed, so they don't cause any extra retains. In my experiments, the `defer` idiom has fewer retain/release. Use `defer`, it's faster.
I'll file a bug for the extra retain/release around calls to fixLifetime in general.

Reply:
Ah, no escape closure captures being guaranteed helps a lot! When did that happen?

Reply:
Recently https://github.com/apple/swift/pull/12957